docs/core-concepts/storage/sdk.mdx

```diff
@@ -1,301 +1,222 @@
+description: File upload, download, and management with the InsForge SDK
 ---
 
 import Installation from '/snippets/sdk-installation.mdx';
 
-InsForge SDK provides simple, bucket-based file storage with automatic URL generation and authentication handling.
-
 <Installation />
 
-## Basic File Operations
+## from()
 
-### Upload a File
+Get a bucket instance for file operations.
 
-<CodeGroup>
-```javascript With Specific Path
-// Upload to specific path in bucket
-const file = fileInput.files[0];
-const { data, error } = await insforge.storage
-  .from('avatars')
-  .upload('user-123/avatar.jpg', file)
+### Parameters
 
-// Returns: { bucket, key, size, mimeType, uploadedAt, url }
-console.log('File URL:', data.url)
-```
+- `bucketName` (string, required) - Name of the storage bucket
 
-```javascript Auto-Generated Name
-// Let the server generate a unique filename
-const file = fileInput.files[0];
-const { data, error } = await insforge.storage
-  .from('avatars')
-  .uploadAuto(file)
+### Returns
 
-// Prevents overwriting existing files
+```typescript
+StorageBucket // Instance with upload, uploadAuto, download, remove methods
 ```
-</CodeGroup>
 
-### Download a File
+### Example
 
 ```javascript
-// Download file as Blob
-const { data: blob, error } = await insforge.storage
-  .from('avatars')
-  .download('user-123/avatar.jpg')
-
-// Create download link
-const url = URL.createObjectURL(blob);
-const a = document.createElement('a');
-a.href = url;
-a.download = 'avatar.jpg';
-a.click();
+const bucket = insforge.storage.from('avatars')
 ```
 
-### List Files
+---
+
+## upload()
 
-```javascript
-// List files in a bucket
-const { data, error } = await insforge.storage
-  .from('avatars')
-  .list({
-    prefix: 'user-123/',  // Optional: filter by prefix
-    limit: 10             // Optional: limit results
-  })
+Upload a file with a specific path/key.
 
-console.log(data.objects)
-// [{ key, size, lastModified, etag, contentType }, ...]
+### Parameters
+
+- `path` (string, required) - The object key/path for the file
+- `file` (File | Blob, required) - File or Blob to upload
+
+### Returns
+
+```typescript
+{
+  data: {
+    bucket: string,
+    key: string,
+    size: number,
+    mimeType: string,
+    uploadedAt: string,
+    url: string
+  } | null,
+  error: Error | null
+}
 ```
 
-### Delete Files
+<Warning>
+If a file with the same key exists, backend auto-renames it. Always use the returned `key` and `url`.
+</Warning>
+
+### Example
 
 ```javascript
-// Delete a single file
-const { error } = await insforge.storage
+const { data, error } = await insforge.storage
   .from('avatars')
-  .remove(['user-123/avatar.jpg'])
+  .upload('user-123/avatar.jpg', fileObject)
 
-// Delete multiple files
-const { error } = await insforge.storage
-  .from('avatars')
-  .remove([
-    'user-123/avatar.jpg',
-    'user-123/cover.jpg'
-  ])
+// Save URL to database
+await insforge.database
+  .from('users')
+  .update({ avatar_url: data.url })
```

Comment:
how to download this image throught url, I thought the coming examples should include this case, but actually not.

Reply:
I am concerning the download.
As a developer, after reading this doc, I know when I uploaded a image I can save the url to related record.
However I don't know how to find it from this doc since there isn't list or query api.
Imagine the scenario of avatar, If I just connect user with the avatar_url, how to update it and how to download it? We can't explain the whole use case clear.